firebase-firestore/src/main/java/com/google/firebase/firestore/core/Query.java

```diff
@@ -325,11 +325,16 @@ public List<OrderBy> getExplicitOrderBy() {
   }
 
   /**
-   * Returns the full list of ordering constraints on the query.
+   * Returns the full list of ordering constraints on the query. This might include additional
+   * sort orders added implicitly to match the backend behavior.
    *
-   * <p>This might include additional sort orders added implicitly to match the backend behavior.
+   * This method is marked as synchronized to prevent the memoizedOrderBy from being
+   * created in multiple threads.
+   *
+   * The returned list is unmodifiable, to prevent ConcurrentModificationExceptions,
```

Comment:
It might be worthwhile to changed the return type, and delete this statement.

Reply:
We can't change the return type since Collection.unmodifiableList(...) returns a List<T>. But I did add one unit test to assert the returned list is unmodifiable.